File: data/readable_format/fortran/file186.f90
module kinds
    implicit none
    integer, parameter :: dp = selected_real_kind(15, 307)
end module kinds

module dot_module
    use kinds, only: dp
    implicit none
contains
    real(kind=dp) function dot(x, y, n)
        integer, intent(in) :: n
        real(kind=dp), intent(in) :: x(n)
        real(kind=dp), intent(in) :: y(n)
        integer :: i
        dot = 0.0_dp
        do i = 1, n
            dot = dot + x(i) * y(i)
        end do
    end function dot
end module dot_module

program test_dot
    use kinds, only: dp
    use dot_module, only: dot
    implicit none
    real(kind=dp) :: result
    real(kind=dp), dimension(3) :: x = [1.0_dp, 2.0_dp, 3.0_dp]
    real(kind=dp), dimension(3) :: y = [4.0_dp, 5.0_dp, 6.0_dp]
    integer :: n = 3

    result = dot(x, y, n)
    print *, "Dot product is:", result
    ! Expected output: Dot product is: 32.0
end program test_dot

File: data/readable_format/cpp/file186.cpp
#include <iostream>
#include <vector>
#include <stdexcept> // For std::invalid_argument
#include <numeric> // For std::inner_product

double dot(const std::vector<double>& x, const std::vector<double>& y);

int main() {
    std::vector<double> x = {1.0, 2.0, 3.0};
    std::vector<double> y = {4.0, 5.0, 6.0};

    try {
        double result = dot(x, y);
        std::cout << "Dot product is: " << result << std::endl;
        // Expected output: Dot product is: 32.0
    } catch (const std::invalid_argument& e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}

double dot(const std::vector<double>& x, const std::vector<double>& y) {
    if (x.size() != y.size()) {
        throw std::invalid_argument("Vectors x and y must be of the same size.");
    }
    double result = std::inner_product(x.begin(), x.end(), y.begin(), 0.0);
    return result;
}